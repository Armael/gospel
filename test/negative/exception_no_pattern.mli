exception E of int

val f : int -> int
(*@ y = f x
    raises E *)
