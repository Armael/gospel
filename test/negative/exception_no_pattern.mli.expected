
*******************************
********** Parsed file ********
*******************************
exception E of int 
val f : int -> int[@@gospel {| y = f x
    raises E |}]

*******************************
****** GOSPEL translation *****
*******************************
(*@ open Stdlib *)

(*@ open Gospelstdlib *)

exception E of int 

val f : int -> int
(*@ y = f x
    with ...
     *)
File "exception_no_pattern.mli", line 5, characters 11-12:
Error: Type checking error: Exception pattern does not match its type.
